java-checks/src/main/java/org/sonar/java/checks/AssertsOnParametersOfPublicMethodCheck.java

```diff
@@ -54,29 +55,22 @@ public void visitNode(Tree tree) {
       return;
     }
     MethodTree methodTree = (MethodTree) tree;
-    if (!methodTree.symbol().isPublic()) {
+    if (!methodTree.symbol().isPublic() || methodTree.symbol().owner().isPrivate()) {
```

Comment:
I would drop that part about private owner. Let's look at the result once deployed on PEACH.